test/cli_test.rb

```diff
@@ -22,18 +21,74 @@ def test_cli_read_config_from_file
 
       # for testing input
       cf = Tempfile.new("test_config")
-      cf.puts <<YAML
-params:
-  run: #{f.path}
-run:
-  - exec: echo hello world >> #{f.path}
-YAML
+      cf.puts <<~YAML
+      params:
+        run: #{f.path}
+      run:
+        - exec: echo hello world >> #{f.path}
+      YAML
       cf.close
 
       Cli.run([cf.path])
       assert_equal("hello world", File.read(f.path).strip)
     end
 
+    def test_cli_gen_docker_run_args_ignores_other_config
+      # When generating the docker run arguments it should ignore other template configuration
+      # like 'run' directives.
+
+      # for testing output
+      f = Tempfile.new("test_output")
+      f.close
+
+      # for testing input
+      cf = Tempfile.new("test_config")
+      cf.puts <<~YAML
+      env:
+        foo: 1
+        bar: 5
+        baz: 'hello_{{spam}}'
+      env_template:
+        spam: 'eggs'
+        config: my_app
+      params:
+        run: #{f.path}
+      run:
+        - exec: echo hello world >> #{f.path}
+      expose:
+        - "2222:22"
+        - "127.0.0.1:20080:80"
+        - 5555
+      volumes:
+        - volume:
+            host: /var/discourse/shared
+            guest: /shared
+        - volume:
+            host: /bar
+            guest: /baz
+      links:
+        - link:
+            name: postgres
+            alias: postgres
+        - link:
+            name: foo
+            alias: bar
+      labels:
+        monitor: "true"
+        app_name: "{{config}}_discourse"
+      YAML
+      cf.close
+
+      expected = []
+      expected << "--env foo=1 --env bar=5 --env baz=hello_eggs"
```

Comment:
This is good, but I want a lot more special chars. 

spaces, new lines, tabs, quotes and so on.